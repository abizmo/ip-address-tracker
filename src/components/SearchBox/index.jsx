function SearchBox() {
  return (
    <input type="text" placeholder="Search for any IP address or domain" />
  );
}

export default SearchBox;
