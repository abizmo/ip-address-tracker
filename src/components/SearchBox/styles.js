import styled from 'styled-components';

export const Form = styled.form`
  display: flex;
  margin-bottom: 1.5rem;
  width: 100%;
`;

export const Input = styled.input`
  border: none;
  border-radius: 1rem 0 0 1rem;
  flex: 1;
  font-family: var(--ff-400);
  font-size: var(--fs-400);
  line-height: 1.1em;
  letter-spacing: 0.015em;
  padding-left: 1.5rem;
  padding-right: 1rem;
`;

export const Submit = styled.button
  .attrs({ type: 'submit' })`
    background-color: var(--clr-900);
    border: none;
    border-radius: 0 1rem 1rem 0;
    height: 58px;
    width: 58px;
  `;
