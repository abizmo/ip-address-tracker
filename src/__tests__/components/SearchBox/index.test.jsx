import { shallow } from 'enzyme';

import SearchBox from '../../../components/SearchBox';

describe('SearchBox test', () => {
  test('should render and match snapshot', () => {
    const wrapper = shallow(<SearchBox />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input')).toBeTruthy();
    expect(wrapper.find('button').prop('type')).toBe('submit');
    expect(wrapper.find('img').prop('src')).toContain('icon-arrow');
  });

  test('should submit when click the button', () => {
    const wrapper = shallow(<SearchBox />);
    const evt = {
      target: { value: '12.12.12.12' },
      preventDefault: () => {},
    };
    jest.spyOn(evt, 'preventDefault');

    wrapper.find('input').simulate('change', evt);
    wrapper.find('form').simulate('submit', evt);
    expect(evt.preventDefault).toBeCalled();
  });
});
