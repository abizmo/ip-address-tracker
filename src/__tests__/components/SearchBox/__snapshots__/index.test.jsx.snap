// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox test should render and match snapshot 1`] = `
<form
  className="sc-bdvvtL fOTIHl"
  onSubmit={[Function]}
>
  <input
    className="sc-gsDKAQ fAohiC"
    id="search"
    onChange={[Function]}
    placeholder="Search for any IP address or domain"
    type="text"
    value=""
  />
  <button
    className="sc-dkPtRN kOHSZC"
    type="submit"
  >
    <img
      alt="submit"
      src="icon-arrow.svg"
    />
  </button>
</form>
`;
